src/graphql/data/TopTokens.ts

```diff
@@ -129,63 +130,101 @@ function toContractInput(token: PrefetchedTopToken) {
   }
 }
 
-export type TopToken = NonNullable<TopTokens_TokensQuery['response']['tokens']>[number]
-interface UseTopTokensReturnValue {
-  loading: boolean
-  tokens: TopToken[]
-  loadMoreTokens: () => void
+// Map of key: ${chain} + ${address} and value: TopToken object.
+// Acts as a local cache.
+const tokensWithPriceHistoryCache: Record<string, TopToken> = {}
+
+const checkIfAllTokensCached = (prefetchedSelectedTokens: PrefetchedTopToken[]) => {
+  let everyTokenHasBeenPreloaded = true
+  const preloadedSelectedTokens: TopToken[] = []
+  prefetchedSelectedTokens.map((token) =>
+    !!token && `${token.chain.toString()}${token.address}` in tokensWithPriceHistoryCache
```

Comment:
I think `.toString` will be called here implicitly, so you don't need to include it.

Reply:
done